### ISSUE TITLE
[nginx-ingress-controller] Upgrading to version 9.3.1  breaks cert-manager SSL certificates

### ISSUE THREAD

**Issue by vudzero** [bitnami/charts#12028]
### Name and Version

bitnami/nginx-ingress-controller

### What steps will reproduce the bug?

Upgrade to version 9.3.1 of the helm chart

### What is the expected behavior?

SSL certificates generated from cert-manager to still works

### What do you see instead?

SSL certificates from cert-manager does not work anymore

### Additional information

I'm getting this error in the nginx controller pod
`2022/08/23 20:19:25 [error] 296#296: *7952 [lua] certificate.lua:259: call(): failed to set DER private key: d2i_PrivateKey_bio() failed, context: ssl_certificate_by_lua*`

And this error in the browser
![image](https://user-images.githubusercontent.com/2973549/186260200-ef5dfb1b-5f58-4dec-bc3f-ce5779177d58.png)

**Comment by vudzero** [comment]
Reverting to 9.3.0 fixes the issue

**Comment by javsalgar** [comment]
Hi!

Thank you for reporting. Could you provide the full steps to reproduce the issue? Including the installation of cert-manager and the certificate creation.

**Comment by steache** [comment]
Hello!
I hope you don't mind if I provide values to reproduce this issue?

We have running `cert-manager` from `jetstack` with this definition: (default chart values used, only overrides in `spec.values`

```
apiVersion: helm.toolkit.fluxcd.io/v2beta1
kind: HelmRelease
metadata:
  name: cert-manager
  namespace: cert-manager
spec:
  releaseName: cert-manager
  interval: 1h0m0s
  chart:
    spec:
      chart: cert-manager
      version: "^1.6.1"
      sourceRef:
        kind: HelmRepository
        name: jetstack
        namespace: flux-system
  values:
    installCRDs: true
    prometheus:
      enabled: false

```

and `nginx-ingress-controller` with this values: (default chart values used, only overrides in `spec.values`)

```
apiVersion: helm.toolkit.fluxcd.io/v2beta1
kind: HelmRelease
metadata:
  name: nginx
  namespace: nginx
spec:
  releaseName: nginx
  interval: 1h0m0s
  chart:
    spec:
      chart: nginx-ingress-controller
      version: "9.3.0"
      sourceRef:
        kind: HelmRepository
        name: bitnami
        namespace: flux-system
  values:
    service:
      externalTrafficPolicy: "Local"
    pdb:
      create: true

```

during auto-update to `bitnami/nginx-ingress-controller` version `9.3.2` an error from the issue.description on top crashed the nginx multiple times :) 

Certificate definition:

```
apiVersion: cert-manager.io/v1
kind: Certificate
metadata:
  name: star-example-com
  namespace: cert-manager
spec:
  secretName: star-example-com
  issuerRef:
    kind: ClusterIssuer
    name: digitalocean-issuer # the problem is also with zerossl-issuer
  commonName: "*.reservio.com"
  dnsNames:
    - "*.example.com"
    - "example.com"
```

[(2022/08/23 20:19:25 [error] 296#296: *7952 [lua] certificate.lua:259: call(): failed to set DER private key: d2i_PrivateKey_bio() failed, context: ssl_certificate_by_lua*)](https://github.com/bitnami/charts/issues/12028#issue-1348514234)

Hope this helps somehow! :)

**Comment by bukington** [comment]
Same issue here, fixed with rollback to 9.3.0

**Comment by kallaics** [comment]
It is probably regression in 9.3.2 again.

Chart version: 9.3.2
Certificates created as kubernetes tls secret object. It is working perfectly.
Nginx reporting the issue above regarding to "DER"

`3427#3427: *21771 [lua] certificate.lua:259: call(): failed to set DER private key: d2i_PrivateKey_bio() failed, context: ssl_certificate_by_lua*,  │
│ client: <IP ADDRESS REDACTED>, server: 0.0.0.0:443`

Please fix it. Thanks.

**Comment by junkiebev** [comment]
Looks like it's _all_ certificates, actually.

**Comment by AlexisDucastel** [comment]
Hi !

Same problem here with 9.3.2 and 9.3.1, we do confirm that 9.3.0 is not concerned by the bug.

error log excerpt: 
```
2022/08/26 12:48:14 [error] 43#43: *1638 [lua] certificate.lua:259: call(): failed to set DER private key: d2i_PrivateKey_bio() failed, context: ssl_certificate_by_lua*, client: XXXXXXXX, server: 0.0.0.0:443
2022/08/26 12:48:17 [error] 43#43: *1784 [lua] certificate.lua:259: call(): failed to set DER private key: d2i_PrivateKey_bio() failed, context: ssl_certificate_by_lua*, client: XXXXXXXX, server: 0.0.0.0:443
```

**Comment by benfoster** [comment]
Ran into the same issue with 9.3.1 using cert manager for both letsencrypt issued certificates and self signed. Roll back to 9.3.0 also worked for me.

**Comment by migruiz4** [comment]
Hi there, 

We have released version 9.3.4 of the `bitnami/nginx-ingress-controller` chart.
Could you please let us know if the issue persists after upgrading?

**Comment by kallaics** [comment]
Hi @migruiz4,
Sorry about the bad news, but I can confirm the issue persists after the upgrade by me.

**Comment by migruiz4** [comment]
Thank you for your feedback.
We will continue working on a fix, sorry for the inconvenience.

**Comment by lknite** [comment]
also seeing, roll back to 9.3.0 got things working again

**Comment by Jeremy-Boyle** [comment]
+1 for seeing this issue

**Comment by PhenixH** [comment]
+1 same persisting issue, only rollback/downgrade  can solve it

**Comment by herrbpl** [comment]
Also reported this to [nginx-ingress controll](https://github.com/kubernetes/ingress-nginx/issues/9002) since I was not sure which part of puzzle is broken.

**Comment by herrbpl** [comment]
I can confirm that using pure https://github.com/kubernetes/ingress-nginx chart ingress-nginx:4.2.3 things work ok. This chart uses nginx-ingress-controller 1.3.0.

**Comment by nofacer** [comment]
same here

**Comment by junkiebev** [comment]
What versions of k8s are y'all running?
I've validated this behavior on 1.19 and 1.20

**Comment by RichieSams** [comment]
1.22

**Comment by cirilloblu** [comment]
1.24

**Comment by kallaics** [comment]
1.22.6

**Comment by tekkamanendless** [comment]
1.21, slowly upgrading toward 1.25 over the next month or so.

**Comment by xavidop** [comment]
same issue with k8s 1.23. nginx version 9.3.9

**Comment by OlegGerber** [comment]
same issue on v1.24.3 and v1.23.9

**Comment by migruiz4** [comment]
Hi everyone,

Just a follow-up that we continue working on this. We will share any updates about a new version fixing this issue here as soon as possible. Thank you for your patience, and sorry for the inconvenience.

**Comment by marcosbc** [comment]
Hi folks, we have just released NGINX Ingress Controller chart 9.3.17, bundling `bitnami/nginx-ingress-controller:1.4.0-debian-11-r2`, which contains a fix for this issue.

Please let us know if it worked for you.

**Comment by AlexisDucastel** [comment]
Hi !

With 9.3.17, the SSL problem is now fixed, but all **our ingresses are in 503** !
We have to rollback to 9.3.0 for now.

Tested on k8s v1.23.7

**Comment by herrbpl** [comment]
Why is Bitnami moving to close-sourced solution? 

Using precompiled results from stacksmith archives, no other party can actually rebuild image and be sure what actual code went into controller binary..

**Comment by kallaics** [comment]
Hi guys,

Unfortunately same issue here.

SSL looks like okay, but the our ingresses are not working well. The response is same error 503.
When I rollback everything works fine.

K8s: 1.22.6 and 1.24.3

**Comment by marcosbc** [comment]
Thanks for your quick responses. We continue looking into this.

**Comment by AlexisDucastel** [comment]
Isn't it a bit soon to close this issue because of a new release 9.3.18 that is not even released yet ?
There have been 16 releases since 9.3.1 that did not fix :) May be we can wait a bit and let some people in this ticket test if it's resolved or not ? Don't you agree @javsalgar ?

**Comment by marcosbc** [comment]
Hi everyone, we have just published version 9.3.18 of the NGINX Ingress Controller chart to the charts registries. We have been able to install it on our side and are not seeing any issues anymore.

If you continue to find any problems with the latest chart version, please let us know.

**Comment by justinmchase** [comment]
Confirmed, it fixes the issue for me.
